### PR TITLE
[WebNN EP] Remove workaround for scalar

### PR DESCRIPTION
**Pull request by Honry**
Currently Chromium has supported scalar with dims = {}, remove legacy workaround for supporting scalar.

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/model_builder.cc [hunk 1]
```diff
@@ -104,13 +104,15 @@ Status ModelBuilder::RegisterInitializers() {
...
+      auto num_elements = SafeInt<size_t>(Product(shape));
       emscripten::val view = emscripten::val::undefined();
       std::byte* tensor_ptr = nullptr;
       if (tensor.has_raw_data()) {
         tensor_ptr = reinterpret_cast<std::byte*>(const_cast<char*>(tensor.raw_data().c_str()));
       } else {
-        std::vector<uint8_t> unpacked_tensor;
+        // Store temporary unpacked_tensor.
+        unpacked_tensors_.push_back({});
+        std::vector<uint8_t>& unpacked_tensor = unpacked_tensors_.back();
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/model_builder.cc:115]: Is this change related, or is it some other fix?

**Review by fdwr — APPROVED** [review]
Nice little cleanup - one less workaround 🧼.

File: onnxruntime/core/providers/webnn/builders/model_builder.cc (hunk 1 above)
Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/model_builder.cc:115]: Scalar needs to be unpacked from tensor, this fixes a bug that previous `unpacked_tensor` is a temporary argument, which will be destructed after this `else` branch.

So we need an global object to store it.

Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/model_builder.cc:115]: It's an existing bug. :p
